content/4_Gold/LIS.mdx

```diff
@@ -144,20 +155,38 @@ int find_lis(vector<int> a) {
 <JavaSection>
 
 ```java
-public static int find_lis(int[] a) {
+public static int findLis(int[] a) {
 	ArrayList<Integer> dp = new ArrayList<Integer>();
 	for (int i : a) {
 		int pos = Collections.binarySearch(dp, i);
-		if(pos < 0) pos = Math.abs(pos + 1);
-		if (pos == dp.size()) dp.add(i);
-		else dp.set(pos, i);
+		if (pos < 0) {
+			pos = Math.abs(pos + 1);
+		}
+		if (pos == dp.size()) {  // we can have a new, longer IS!
+			dp.add(i);
+		} else {  // oh ok, at least we can make the ending element smaller
+			dp.set(pos, i);
+		}
```

Comment:
Seems unnecessarily long compared to the 3 concise lines D:

Reply:
They're equally concise, though, because the code is the same. One just makes control flow more obvious.

Reply:
It's a single statement though. I'd argue that making the code more vertically concise is better for readability because it's more compact and elegant

Reply:
I mean with monitors as big as they are, being vertically concise shouldn't be that big of a concern (also I compacted the `Math.abs` part).

Reply:
> more compact and elegant

No, the former code is not, in fact, more elegant.

Reply:
How so?

Reply:
> control flow more obvious

Yes, in the original way you have a series of (condition -> statement). This exactly contributes to why concise is cleaner.

The alternative requires you to connect condition & statement between multiple lines. Of course this would be preferred for a set of substantial, multiline statement, but not when it is a sleek one-liner.

Reply:
In the end settling with the choice of the original author is just more respectful, especially when there is no significant reason to change.

Reply:
By virtue of the semantics being the exact same. The indentation present in the latter, however, makes the control flow easier to digest—that is the point of indentation. Braces are a good practice anyway and doesn't make it any less "elegant" or readable.